openedx/features/djangoapps/calendar_tab/views.py

```diff
@@ -0,0 +1,180 @@
+            publish_calendar(created_calendar['id'])
+        except Exception as e:
+            # TODO: handle errors
+            print(e)
+            return JsonResponse({"errors": []}, status=400)
+        else:
+            updated = CourseOverview.objects.filter(id=course_key) \
+                                            .update(calendar_id=created_calendar['id'])
+            return JsonResponse({"calendarId": created_calendar['id']}, status=201)
+
+
+class CalendarView(TemplateView):
+    """Main view: renders calendar"""
+    template_name = 'calendar_tab/calendar_tab_page.html'
+
+    def get_context_data(self, **kwargs):
+        course_key = CourseKey.from_string(self.kwargs['course_id'])
+        course = get_course_with_access(self.request.user, "load", course_key)
+        is_staff = bool(has_access(self.request.user, 'staff', course))
+        calendar_id = get_calendar_id_by_course_id(course_key)
+
+        context = super(CalendarView, self).get_context_data(**kwargs)
+        context['course'] = course
+        context['is_staff'] = is_staff
+        context['calendar_id'] = calendar_id
+        return context
+
+
+def events_view(request, course_id):
+    """Returns all google calendar events for given course"""
+    course_key = CourseKey.from_string(course_id)
+    calendar_id = get_calendar_id_by_course_id(course_key)
+    try:
+        response = gcal_service.events().list(calendarId=calendar_id, pageToken=None).execute()
+        events = [{
+                      "id": event["id"],
+                      "text": event["summary"],
+                      "start_date": from_google_datetime(event["start"]["dateTime"]),
+                      "end_date": from_google_datetime(event["end"]["dateTime"])
+                  } for event in response['items']]
+    except Exception as e:
+        # TODO: handle errors
+        print(e)
+        return JsonResponse(data={'errors': e}, status=500, safe=False)
+    else:
+        return JsonResponse(data=events, status=200, safe=False)
+
+
+@csrf_exempt
+def dataprocessor_view(request, course_id):
+    """Processes insert/update/delete event requests"""
+    course_key = CourseKey.from_string(course_id)
+    calendar_id = get_calendar_id_by_course_id(course_key)
+    status = 401
+    response = {'action': 'error',
+                'sid': request.POST['id'],
+                'tid': '0'}
+
+    def get_event_data(post_data):
+        event = {
+            'id': post_data.get('id'),
+            'summary': post_data['text'],
+            'location': post_data.get('location') or '',
+            'description': post_data.get('description') or '',
+            'start': {
+                'dateTime': to_google_datetime(post_data['start_date']),
+            },
+            'end': {
+                'dateTime': to_google_datetime(post_data['end_date']),
+            },
+        }
+        return event
+
+    if request.method == 'POST':
```

Comment:
So is this the only way to interact with `dhtmlxscheduler`?
I prefer to have separate entry points a-la REST, rather than single entry point for everything.